### DISCUSSION TITLE
video.js

### DISCUSSION THREAD

**Discussion by astelmack** [muxinc/media-chrome#208] [Q&A] [answered]
Out of curiosity, is it possible the have media-chrome manage the chrome and functionality for a video.js player?  Not sure if this is a silly question, but just seems like the architecture around the UI might be nicer here.

**Comment by heff** [comment]
Hey, not a silly question. There's not a way I know of today, but @gkatsev or @mmcc might have ideas. We just need a custom element wrapper of video.js that translates the video element API to the video.js API.

We've had some conversations about splitting out "videojs-core" as just the playback logic, which would then make more sense to wrap.

**Reply by gkatsev** [reply]
At first, I thought that it would be an issue to wrap Video.js in a custom element due to some assumptions that Video.js makes around the shadow DOM, but it's possible that wouldn't make that much of a difference since most of those assumptions are around its controls.

Splitting up Video.js into more reusable pieces would be great, but a larger undertaking. Though, the custom element could be updated to use that when and if that happens.

**Comment by luwes** [comment, marked as answer]
This should get you on the path of using Media Chrome for a video.js player. let me know if you hit any bugs. 🙏 

https://github.com/luwes/videojs-video-element
https://github.com/muxinc/media-chrome/compare/main...luwes:videojs?expand=1

**Reply by gkatsev** [reply]
Cool! Though you beat me to this; I was going to look at this next!
I'll have to take a look, interested to see how you set it up.
